medallion/backends/memory_backend.py

```diff
@@ -122,14 +123,37 @@ def _pop_expired_sessions(self):
 
         for item in expired_ids:
             self.next.pop(item)
-
+    #checks collections for proper manifest, if objects are present in a collection, a manifest should be present with 
+    #an entry for each entry in objects
+    def collections_manifest_check(self):
+        for group in self.data:
+            if 'collections' in self.data[group].keys():
+                collections = self.data[group]['collections']
+                if collections:
+                    for collection in collections:
+                        if collection['objects']:
+                            if 'manifest' not in collection:
+                                raise InitializationError("Collection {} manifest is missing".format(collection['id']), 408)
+                            if not collection['manifest']:
+                                raise InitializationError("Collection {} with objects has an empty manifest".format(collection['id']), 408)
+                            for obj in collection['objects']:
+                                obj_time=find_att(obj)
+                                obj_man_paired = False
+                                for man in collection['manifest']:
+                                    man_time = find_att(man)
+                                    if obj['id'] == man['id'] and obj_time == man_time:
+                                        obj_man_paired = True
+                                if not obj_man_paired: 
+                                    raise InitializationError("Object with id {} last modified on {} is missing a manifest".format(obj['id'], obj_time), 408)
```

Comment:
This much nesting is kind of a code smell. We could simplify it using `whatever_dict.get('whatever_key', {})`. This removes the need for checking if a key is in the dictionary - if it's not, `.get()` will use the default we give it, which could be an empty dictionary or empty list. So we could use `for collection in root.get('collections', [])` instead of lines 130-133. Then we can do something similar for objects inside the collection.